### PR TITLE
Add CRUD for Games model

### PR DESCRIPTION
**Pull request by erinmorrissey**
So this shows 21 changed files, but that's b/c this PR is dependent on the DB PR — does the number of files changed automatically adjust when that PR is merged into master, so that the diffs are easier to review? There should actually be only ~6 changed files after that happens...

Dependent on:
- [x] #5

### CODE REVIEW

**Comment by erinmorrissey** [comment]
dependent on PR #5

**Comment by lasley** [comment]
Yeah the file diff is a dynamic differential between this branch and the one you're requesting merge to. The changes will update as master is.

In real world repos you would also rebase your branch on master, which would remove the commits too, but we won't do that here.

**Comment by erinmorrissey** [comment]
Cool, then this is ready for review. I'll need to clean up the tests with Factory Girl, but will do that after some of the file diffs are resolved.

**Comment by erniejamison** [comment]
LGTM 👍

File: app/controllers/games_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class GamesController < ApplicationController
```
Comment (lasley) [inline app/controllers/games_controller.rb:1]: Missing the U & D from CRUD (update/delete)

https://github.com/tjh/Basic-Rails-CRUD/blob/master/app/controllers/posts_controller.rb

Comment (lasley) [inline app/controllers/games_controller.rb:1]: I'd be good with leaving out delete if you want. We'll need update though

**Comment by lasley** [comment]
@erinmorrissey - so it seems that your last commit contains both your code + a merge. Merges need to be independent from your work, or we will end up with conflicts. 

You can see what this looks like in the network graph - https://github.com/teamdeepblue/caissa-chess/network

image

Note how your commit also has a line coming from some other branch. This means it is also a merge & has two parents. With two parents, things get sketchy in terms of determining what was done - it is assumed any diff was conflict resolution work.

I will fix this up in another branch and PM the name to you so that you can confirm I didn't screw anything up. Some tricky gittery is going to be used & I don't want to accidentally delete any of your work.

**Comment by lasley** [comment]
New branch is rebase/crud-for-games

https://github.com/teamdeepblue/caissa-chess/compare/rebase/crud-for-games

Let me know if that looks right and I'll move it to this branch.

**Comment by erinmorrissey** [comment]
@lasley Yes, that looks right. So if I grab a branch other than master to base my new work off of, do I want to make sure only the files I actually changed are committed and pushed? I'm in the habit of doing a 'git add --all' before making commits.

**Comment by lasley** [comment]
A merge commit is a bit different than normal. You should only include work related to merge conflicts in those commits.

That said, a `git add -a` probably didn't do that here. I think what instead happened was you forgot to actually commit the merge, then continued to work & eventually committed later. I used to do it all the time, so that's my guess 😛 

Alright so I'm going to rebase my branch onto this one. This will desync your local, so you'll need to do something like:

```
git checkout crud-for-games
git reset --hard 61e95fa1a35987042abbe6c1cf1fc614b7a2f6d3
```

File: app/controllers/games_controller.rb (hunk 1 above)
Comment (erinmorrissey) [reply app/controllers/games_controller.rb:1]: @lasley So if I add an Update method right now, all that can be updated is the game :name. I would assert there isn't a reason to allow the name to be updated, and in that case, I can't think what we would need an Update method for at this point...?

**Comment by lasley** [comment]
LGTM 👍

File: app/controllers/games_controller.rb (hunk 1 above)
Comment (IlianaP) [reply app/controllers/games_controller.rb:1]: @erinmorrissey I actually already created the update method in the UpdateAction branch... I guess we have been working on the same thing...I know that the update method was part of CRUD and I was doing the "add player to a game", so I guess there was overlap

**Pull request merged**